src/utils/parse_api_token.py

```diff
@@ -0,0 +1,26 @@
+from src.utils.logger import Log, logger
+from py.path import local
+from typing import Tuple, Union
+import json
+
+
+@Log(logger)
+def parse_api_token(file: Union[str, local], key_username: str = "username",
+                    key_token: str = "token") -> Tuple[str, str]:
+    """Parse a JSON file containing an API username and token.
+
+    Args:
+        file: A JSON file containing a username and token, where the keys for each are key_username, and key_token.
+        key_username: The username key in file.
+        key_token: The token key in file.
+
+    Returns:
+        A tuple of two elements, where the first element is the username, and the second element is the token.
+
+    """
+
+    # Read and parse the JSON file, and return only the key_username, and key_token keys in the file
+    with open(file, "r") as f:
```

Comment:
It's not clear to me why we're not re-using the `.secrets` file to read in these values? I worry that these credentials would be accidentally committed in an unknown JSON file.

Reply:
That's a very good point @chao-xian! I think by trying to add extra security, I've introduced a flaw! 😬 

I've made changes to the code in commit [`33db318`](https://github.com/ukgovdatascience/github_organisation_administration/pull/1/commits/33db31875d366a26a677c27fbd721e8eae295769) to remove the `parse_api_token`, and all its references, and revised the documentation to state that the API token should be added as an environment variable in the `.secrets` file. Let me know what you think!